common/crypto/verify_ias_report/verify-report.h

```diff
@@ -18,34 +18,57 @@
 
 #include <sgx_quote.h>
 
-typedef enum {
+typedef enum
+{
     VERIFY_SUCCESS,
     VERIFY_FAILURE
 } verify_status_t;
 
+typedef enum
+{
+    QS_INVALID,
+    QS_OK,
+    QS_GROUP_OUT_OF_DATE,
+    QS_CONFIGURATION_NEEDED,
+    QS_SW_HARDENING_NEEDED,
+    QS_CONFIGURATION_AND_SW_HARDENING_NEEDED,
+    QS_NUMBER
+} quote_status_e;
+
+#define QSF_ACCEPT_GROUP_OUT_OF_DATE (1 << QS_GROUP_OUT_OF_DATE)
+#define QSF_ACCEPT_CONFIGURATION_NEEDED (1 << QS_CONFIGURATION_NEEDED)
+#define QSF_ACCEPT_SW_HARDENING_NEEDED (1 << QS_SW_HARDENING_NEEDED)
+#define QSF_ACCEPT_CONFIGURATION_AND_SW_HARDENING_NEEDED \
+    (1 << QS_CONFIGURATION_AND_SW_HARDENING_NEEDED)
+#define QSF_ACCEPT_ALL UINT_MAX
+#define QSF_REJECT_ALL (0)
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 int get_quote_from_report(const uint8_t* report, const int report_len, sgx_quote_t* quote);
-verify_status_t verify_enclave_quote_status(const char* ias_report, int ias_report_len, int group_out_of_date_is_ok);
+verify_status_t verify_enclave_quote_status(
+    const char* ias_report, unsigned int ias_report_len, unsigned int quote_status_flags);
 verify_status_t verify_ias_certificate_chain(const char* cert_pem);
 verify_status_t verify_ias_report_signature(const char* ias_attestation_signing_cert_pem,
-                                            const char* ias_report,
-                                            unsigned int ias_report_len,
-                                            char* ias_signature,
-                                            unsigned int ias_signature_len);
+    const char* ias_report,
+    unsigned int ias_report_len,
+    char* ias_signature,
+    unsigned int ias_signature_len);
+
+quote_status_e get_quote_status(const char* ias_report, unsigned int ias_report_len);
 #ifdef __cplusplus
 }
 #endif
 
-#define COND2ERR(b) \
-    do \
-    { \
-        if(b) \
-        { \
+#define COND2ERR(b)   \
+    do                \
+    {                 \
+        if (b)        \
+        {             \
             goto err; \
```

Comment:
any particular reason we aren't using the throw constructs that are defined in common and used pretty much every where else?

Reply:
To avoid handling exceptions, since this particular piece of code could in theory be compiled as pure C code, and this is desirable when used in CGO.

Reply:
have you tried the full compilation in C mode? should CONDERR be added to one of the logging/error header files so that it can become "canonical"?